x/sequencers/keeper/msg_server.go

```diff
@@ -113,3 +118,91 @@ func (m msgServer) UpsertSequencer(goCtx context.Context, msg *types.ConsensusMs
 
 	return &types.ConsensusMsgUpsertSequencerResponse{}, nil
 }
+
+// defines the list of accounts we want to bump the sequence
+var handleAccounts = map[string]struct{}{
+	proto.MessageName(&authtypes.BaseAccount{}):                 {},
+	proto.MessageName(&vestingtypes.BaseVestingAccount{}):       {},
+	proto.MessageName(&vestingtypes.ContinuousVestingAccount{}): {},
+	proto.MessageName(&vestingtypes.DelayedVestingAccount{}):    {},
+	proto.MessageName(&vestingtypes.PeriodicVestingAccount{}):   {},
+	proto.MessageName(&vestingtypes.PermanentLockedAccount{}):   {},
+}
+
+const BumpSequence = 10_000_000_000
+
+func (m msgServer) BumpAccountSequences(goCtx context.Context, msg *types.MsgBumpAccountSequences) (*types.MsgBumpAccountSequencesResponse, error) {
+	if msg.Authority != m.authority {
+		return nil, sdkerrors.ErrorInvalidSigner.Wrapf("only an authorized actor can bump account sequences")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	var err error
+	m.accountKeeper.IterateAccounts(ctx, func(account authtypes.AccountI) bool {
+		accType := proto.MessageName(account)
+		_, ok := handleAccounts[accType]
+		i := 0
+		for !ok && i < len(m.accountBumpFilters) {
```

Comment:
if a filter returns ok=true, the loop still continues instead of breaking and calling bumpAccountSequence

Reply:
the loop is `for !ok..`